src/common/k8s-api/json-api.ts

```diff
@@ -65,22 +69,33 @@ export class JsonApi<D = JsonApiData, P extends JsonApiParams = JsonApiParams> {
     debug: false
   };
 
-  constructor(public readonly config: JsonApiConfig, protected reqInit?: RequestInit) {
+  constructor(public readonly config: JsonApiConfig, protected reqInit?: JsonRequestInit) {
     this.config = Object.assign({}, JsonApi.configDefault, config);
     this.reqInit = merge({}, JsonApi.reqInitDefault, reqInit);
     this.parseResponse = this.parseResponse.bind(this);
   }
 
-  public onData = new EventEmitter<[D, Response]>();
-  public onError = new EventEmitter<[JsonApiErrorParsed, Response]>();
+  public onData = new EventEmitter<[D, JsonResponse]>();
+  public onError = new EventEmitter<[JsonApiErrorParsed, JsonResponse]>();
 
-  get<T = D>(path: string, params?: P, reqInit: RequestInit = {}) {
+  get<T = D>(path: string, params?: P, reqInit: RequestInit | NodeRequestInit = {}) {
     return this.request<T>(path, params, { ...reqInit, method: "get" });
   }
 
-  getResponse(path: string, params?: P, init: RequestInit = {}): Promise<Response> {
-    let reqUrl = `${this.config.serverAddress}${this.config.apiBase}${path}`;
-    const reqInit: RequestInit = merge({}, this.reqInit, init);
+  protected getRequestUrl(path: string, useSubdomain = false) {
+    if (window) {
```

Comment:
I guess it's not gonna work in node env. 
You cannot check global variable existence this way, use `typeof` at least:

```
% node -i
Welcome to Node.js v14.17.6.
Type ".help" for more information.
> window
Uncaught ReferenceError: window is not defined
> typeof window
'undefined'
```